src/dune_rules/simple_rules.ml

```diff
@@ -227,6 +227,8 @@ let copy_files sctx ~dir ~expander ~src_dir (def : Copy_files.t) =
     in
     Build_system.eval_pred (File_selector.of_glob ~dir glob)
   in
+  if Filename_set.is_empty files
```

Comment:
This has to be guarded by a version check since older versions can rely on the old behavior.